src/Xamarin.Android.Build.Tasks/Tasks/GenerateILLinkXml.cs

```diff
@@ -0,0 +1,70 @@
+using System.Xml.Linq;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks;
+
+public class GenerateILLinkXml : AndroidTask
+{
+	public override string TaskPrefix => "GILX";
+
+	[Required]
+	public string AndroidHttpClientHandlerType { get; set; }
+
+	[Required]
+	public string CustomViewMapFile { get; set; }
+
+	[Required]
+	public string OutputFile { get; set; }
+
+	public override bool RunTask ()
+	{
+		string assemblyName, typeName;
+
+		var index = AndroidHttpClientHandlerType.IndexOf (',');
+		if (index != -1) {
+			typeName = AndroidHttpClientHandlerType.Substring (0, index).Trim ();
+			assemblyName = AndroidHttpClientHandlerType.Substring (index + 1).Trim ();
+		} else {
+			typeName = AndroidHttpClientHandlerType;
+			assemblyName = "Mono.Android";
+		}
+
+		// public parameterless constructors
+		// example: https://github.com/dotnet/runtime/blob/039d2ecb46687e89337d6d629c295687cfe226be/src/mono/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml
+		var ctor = new XElement ("method", new XAttribute("signature", "System.Void .ctor()"));
+
+		XElement linker;
+		var doc = new XDocument (
+			linker = new XElement ("linker",
+				new XElement ("assembly",
+					new XAttribute ("fullname", assemblyName),
+					new XElement ("type", new XAttribute ("fullname", typeName), ctor)
+				)
+			)
+		);
+
+		var customViewMap = MonoAndroidHelper.LoadCustomViewMapFile (BuildEngine4, CustomViewMapFile);
+		foreach (var pair in customViewMap) {
+			index = pair.Key.IndexOf (',');
+			if (index == -1)
+				continue;
+
+			typeName = pair.Key.Substring (0, index).Trim ();
+			assemblyName = pair.Key.Substring (index + 1).Trim ();
+
+			linker.Add (new XElement ("assembly",
+				new XAttribute ("fullname", assemblyName),
+				new XElement ("type", new XAttribute ("fullname", typeName), ctor)
+			));
+		}
```

Comment:
This code doesn't work yet, as the file contains:
```
mono.android_test.library.CustomTextView;D:\src\xamarin-android\tests\Mono.Android-Tests\Runtime-Microsoft.Android.Sdk\obj\Release\net9.0-android\lp\6\jl\res\layout\upper_lower_custom.xml
mono.android_test.library.CustomTextView;obj\Release\net9.0-android\res\layout\upper_lower_custom.xml
Mono.Android_Test.Library.CustomTextView;D:\src\xamarin-android\tests\Mono.Android-Tests\Runtime-Microsoft.Android.Sdk\obj\Release\net9.0-android\lp\6\jl\res\layout\upper_lower_custom.xml
Mono.Android_Test.Library.CustomTextView;obj\Release\net9.0-android\res\layout\upper_lower_custom.xml
Mono.Android_Test.Library.CustomTextView;obj\Release\net9.0-android\res\layout\uppercase_custom.xml
```

I need the full type name like `Mono.Android_Test.Library.CustomTextView, Mono.Android_Test.Library`.

Reply:
Using the acw_map.txt file might allow you to map that info back. You will probably need a new file though.

Reply:
I tried to piece together the information with existing files:
* `customview-map.txt` has an entry with a type name and file name like `upper_lower_custom.xml`
* `<ConvertResourcesCases/>` has `$(MonoAndroidResDirIntermediate);@(_LibraryResourceDirectories)`
* `@(_LibraryResourceDirectories)` has `OriginalFile = D:\src\xamarin-android\tests\Mono.Android-Tests\Mono.Android-Test.Library\bin\Release\net9.0-android\Mono.Android-Test.Library.NET.aar`
* `Mono.Android-Test.Library.NET.aar` is created based on `AndroidResource` files in `Mono.Android-Test.Library.NET.dll`

So, it doesn't seem like we even have data to map back to the assembly name. I'll keep thinking about this one.